src/main/javascript/src/pages/privilege/create/create.js

```diff
@@ -35,6 +35,6 @@ export class setPrivilegeFormFields {
 			owner: ownerChoice,
 			operation: operationChoice
 		}
-		this.privilegeManager.addPrivilege(this.myPrivilege).then(()=>{this.router.navigateToRoute("view roles")});
+		this.privilegeManager.addPrivilege(this.myPrivilege);
```

Comment:
Always need a .then() to display feedback and a .catch() for errors otherwise this provides us with no feedback and nothing updates in the front end so we don't know if anything happened until we refresh, which is an issue in almost every element of this PR.